### PR TITLE
Ft import and save many trainees from googlesheet v5

### PR DESCRIPTION
**Pull request by Niyonkuru1**
## What does this PR do?

Import many trainees and their attributes from google sheet and save them into the mongo database

## Description of Task to be completed?

The admin should copy the URL of his / her google sheet and paste them into the input field of the app, then after hitting load, those data should be copied into the database.

 If the column of the google sheet is not matching the properties in the database collection, he/she should see the error screen with the steps of how to do it correctly. while if it save correctly, he / she will see the screen saying that uploading is successful.

## How should this be manually tested?

- `Clone the repo`

- `checkout` to branch `ft-import-and-save-many-trainees-from-googlesheet-v5`

- Pull the changes by `git pull`

- Install all packages by running `npm install`

- Start app by `running npm run dev`

## Any background context you want to provide?

Note note **you should use only the configured google sheet url, not every google sheet url will work.**
please use This google sheet to test:"https://docs.google.com/spreadsheets/d/1hXUavm_K5BQAOGTx9W4Z84Ttb2_ONfvk_EQZlyOnUPM/edit#gid=0"

`How to use this link in front end`
=>When you are on filter_trainee page you will found import from (button) and when you hint this you get form.

=>Just copy the whole link and paste to provided input and hint reload data to database.

Screen Shot 2022-11-16 at 16 44 07

And when you get an other form tell you "Please enter the correct cycle name which is in Database already!!!'
Screen Shot 2022-11-16 at 16 44 43

=>Then provide  the circle id which is in the database and hint button (Then send your query again)
When all things are collect you will be redirected to filter trainee

Screen Shot 2022-11-16 at 16 44 46

## What are the relevant pivotal tracker stories?

[Trello Board reference](https://trello.com/c/17S8aqBu/58-import-trainees-google-sheet)

### CODE REVIEW

File: src/redux/app/hook.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
```
Comment (capitale250) [inline src/redux/app/hook.ts:1]: we was having this hook src/hooks/hooks.ts

File: src/redux/actions/PerformLoadDataAction.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  interview_decision: "",
+  past_andela_programs: "",
+  cycle_name: "",
+};
+
+export const loadDataIntoDb = (googleSheetId: string) => {
+  return async (dispatch: any) => {
+    dispatch(load_data_request());
+    const resultPromise = axios({
+      url: "http://localhost:4000/",
```
Comment (UmuhireJessie) [inline src/redux/actions/PerformLoadDataAction.ts:37]: Change `url: "http://localhost:4000/",` to `url: process.env.BACKEND_URL,`

File: src/redux/actions/PerformLoadDataAction.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    if ((await resultPromise).data.data) {
+      console.log((await resultPromise).data.data.loadAllTrainees);
+      toast.success(`Saved successfully 👌`)
+      dispatch(
+        load_data_success((await resultPromise).data.data.loadAllTrainees)
+      );
+    }
+    else {
+      console.log((await resultPromise).data.errors[0].message);
+      toast.error(`OOPS!!! import failed please match columns 🤯`)
```
Comment (UmuhireJessie) [inline src/redux/actions/PerformLoadDataAction.ts:61]: I think you might consider removing the emoji

File: src/redux/actions/PerformLoadDataAction.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        toast.success(`Saved  import successfully 👌`)
+        console.log((await resultPromise).data.data.reSendDataIntoDb);
+        dispatch(
+          load_data_success((await resultPromise).data.data.reSendDataIntoDb)
+        );
+      }
+
+      // on errors
+      else {
+        toast.error(`Not saved please match cohort name !! 🤯`)
```
Comment (UmuhireJessie) [inline src/redux/actions/PerformLoadDataAction.ts:101]: same here

File: src/pages/importAndSaveManyTraineesFromGoogleSheet/importAndSaveManyTraineesFromGoogleSheet.tsx [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+          {/* <label htmlFor="" className="">Please enter the correct cycle name which is in Database already!!!</label>
+        <input type="text" /> */}
+
+          <form onSubmit={handleSubmitToResend}>
+            <div className="mb-6">
+              <label
+                htmlFor="text"
+                className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
+              >
+                Please enter the correct cycle name which is in Database
```
Comment (UmuhireJessie) [inline src/pages/importAndSaveManyTraineesFromGoogleSheet/importAndSaveManyTraineesFromGoogleSheet.tsx:165]: Change the message to something simple as @denislohan suggested.
Ex:  `Please enter correct cycle name!`

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
So far so good! 

1. You might have deleted this toast that shows that importing trainees is being processed, please add it again. Otherwise, a user can not know what's happening 
image

2. Please consider the suggested changes on the backend too

3. Find a way to save trainee applicants if all checks of trainee attributes are finished. Otherwise `/trainee` will have more trainees compared to `/filter_trainee` page which is a problem.

**Pull request merged**